javascript/ql/src/experimental/Security/CWE-94/ServerSideTemplateInjection.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @name Server Side Template Injection
+ * @description Rendering templates with unsanitized user input allows a malicious user arbitrary
+ *              code execution.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id js/code-injection
+ * @tags security
+ *       external/cwe/cwe-094
+ */
+
+import javascript
+import DataFlow
+
+class ServerSideTemplateInjectionConfiguration extends TaintTracking::Configuration {
+  ServerSideTemplateInjectionConfiguration() { this = "ServerSideTemplateInjectionConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof ServerSideTemplateInjectionSink }
+}
+
+abstract class ServerSideTemplateInjectionSink extends DataFlow::Node { }
+
+class SSTIPugSink extends ServerSideTemplateInjectionSink {
+  SSTIPugSink() {
+    exists(CallNode compile, ModuleImportNode renderImport |
+      renderImport = moduleImport(["pug", "jade"]) and
+      (
+        compile = renderImport.getAMemberCall("compile") and
+        exists(compile.getACall())
+        or
```

Comment:
```suggestion
        compile = renderImport.getAMemberCall("compile") and
        or
```
Do we need the `exists(compile.getACall())` restriction at all? 
Does it ever happen that someone calls `compile()` without eventually without eventually invoking the compiled template? 
(If it does happen, then feel free to drop this suggestion)

Reply:
> Do we need the exists(compile.getACall()) restriction at all?
Does it ever happen that someone calls compile() without eventually without eventually invoking the compiled template? 

I've been thinking about it and no, there is no usage of `compile()` except invocation of its result (`compile()` returns function). So lets assume it will be invoked.